woocommerce/woocommerce-gutenberg-products-block
# Add `CurrencyCode` type for the Currency `code` property

Currently, the currency interface accepts any string as currency code, which would allow entering incorrect values.

This PR adds a new `type CurrencyCode` used as the type for the `code` and `currency_code` properties. That way the possible string values are restricted only to the allowed _ISO 4217_ codes.

Fixes https://github.com/woocommerce/woocommerce-gutenberg-products-block/issues/5397

### Testing

#### Automated Tests

### Manual Testing
1. Create a new page with the "All Products" and the "Filter Products by Price" blocks.
2. Go to the page on the front
3. Check that the price/currency is showing ok on the slider inputs and the product prices.

-

1. Add a product to the cart
2. Go to the checkout
3. Check the price and currency are ok
